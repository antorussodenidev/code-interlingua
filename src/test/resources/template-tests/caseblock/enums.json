{
  "type": "java.lang.annotation.RetentionType",
  "cases": [
    {
      "value": { "nodeType": "EnumConstantName", "code": "FIELD" },
      "stmts": [
        { "nodeType": "Statement", "code": "foo(x);" },
        { "nodeType": "Statement", "code": "bar();" }
      ]
    },
    {
      "value": { "nodeType": "EnumConstantName", "code": "METHOD" },
      "stmts": [
        { "nodeType": "Statement", "code": "baz();" }
      ]
    }
  ]
}
