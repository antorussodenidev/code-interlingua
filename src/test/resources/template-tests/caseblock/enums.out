package foo;

class C {
  C(java.lang.annotation.RetentionType x) {
    switch (x) {
      case FIELD :
        foo(x);
        bar();
        break;
      case METHOD :
        baz();
        break;
    }
  }
}